src/account/NexusSmartAccount.ts

```diff
@@ -2073,16 +2078,18 @@ export class NexusSmartAccount extends BaseSmartContractAccount {
    * @param moduleType - The type of module to check for support.
    * @returns A promise that resolves to a boolean indicating whether the module type is supported.
    */
-  async supportsModule(module: Module): Promise<boolean> {
+  async supportsModule(moduleType: ModuleType): Promise<boolean> {
     const accountContract = await this._getAccountContract()
     const moduleIndex =
-      module.type === "validator"
+    moduleType === "validator"
         ? 1n
-        : module.type === "executor"
+        : moduleType === "executor"
           ? 2n
-          : module.type === "fallback"
+          : moduleType === "fallback"
             ? 3n
-            : 4n
+            : moduleType === "hook"
+              ? 4n
+              : 0n // review
```

Comment:
default should be 0?
there may be more module types in the future.
in consensus with ERC7579